packages/cozy-konnector-libs/src/libs/saveBills.js

```diff
@@ -197,4 +199,36 @@ function checkRequiredAttributes(entries) {
     }
   }
 }
+
+function appendContractsDatas(tempEntries, options) {
+  // if contractId passed by option
+  if (options.contractId) {
+    // Surcharge label if not define
+    if (!options.contractLabel) {
+      options.contractLabel = options.contractId
+    }
+    options.subPath = options.contractLabel
+    // Add datas in bills
+    for (let entry of tempEntries) {
+      entry.contractLabel = options.contractLabel
+      entry.contractId = options.contractId
+    }
+    return tempEntries
+  } else if (options.contractLabel) {
+    log('warn', 'contractLabel used without contractId')
+  }
+
+  // if contractId passed by bill
+  for (let entry of tempEntries) {
+    if (entry.contractId) {
+      if (!entry.contractLabel) {
+        entry.contractLabel = entry.contractId
+      }
+      entry.subPath = entry.contractLabel
+    }
+  }
+  return tempEntries
+}
```

Comment:
IMHO there is too much stuff in this function.

- Defaulting options
- Iteration
- Changing attributes
- 2 returns

Do you think you could split this function up?

Reply:
Solve by splitting in https://github.com/konnectors/libs/commit/3bb9d21e50ae9a4f368c0059b28639992ec623e9